pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -467,3 +479,77 @@ func containsCondition(target k6tv1.VirtualMachineConditionType, elems []k6tv1.V
 	}
 	return false
 }
+
+func CollectDiskAllocatedSize(vms []*k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
+	var cr []operatormetrics.CollectorResult
+
+	for _, vm := range vms {
+		if vm.Spec.Template == nil {
+			continue
+		}
+		cr = append(cr, collectDiskMetricsFromVMVolumes(vm)...)
+	}
+
+	log.Log.Infof("Returning result set: %+v", cr)
+	return cr
+}
+
+func collectDiskMetricsFromVMVolumes(vm *k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
+	var cr []operatormetrics.CollectorResult
+
+	for _, vol := range vm.Spec.Template.Spec.Volumes {
+		if vol.PersistentVolumeClaim != nil {
+			diskName := vol.Name
+			pvcName := vol.PersistentVolumeClaim.ClaimName
+
+			key := fmt.Sprintf("%s/%s", vm.Namespace, pvcName)
+			obj, exists, err := persistentVolumeClaimInformer.GetStore().GetByKey(key)
+			if err != nil || !exists {
+				cr = append(cr, collectDiskMetricsFromList(vm, pvcName, diskName)...)
+			} else {
+				pvc, ok := obj.(*k8sv1.PersistentVolumeClaim)
+				if ok {
+					pvcSize := pvc.Spec.Resources.Requests.Storage()
+					volumeMode := "null"
+					if pvc.Spec.VolumeMode != nil {
+						volumeMode = string(*pvc.Spec.VolumeMode)
+					}
+					cr = append(cr, operatormetrics.CollectorResult{
+						Metric: vmDiskAllocatedSize,
+						Value:  float64(pvcSize.Value()),
+						Labels: []string{vm.Name, vm.Namespace, pvcName, volumeMode, diskName},
+					})
+				}
+			}
+		}
+	}
+
+	return cr
+}
+
+func collectDiskMetricsFromList(vm *k6tv1.VirtualMachine, pvcName, diskName string) []operatormetrics.CollectorResult {
+	var cr []operatormetrics.CollectorResult
+
+	pvcList := persistentVolumeClaimInformer.GetIndexer().List()
+	if len(pvcList) == 0 {
+		return cr
+	}
+
+	for _, obj := range pvcList {
```

Comment:
Why going through this list to find a single PVC when you can call `.GetStore().GetByKey("namespace/name")`

Reply:
I'm doing it in this line:https://github.com/kubevirt/kubevirt/pull/12765/files#diff-1f9be9d6f2363d168e35db520321088d1d544b5c7afd205f177d002f46f13129R506

not sure why here no, I will fix

Reply:
fixed to use one func that uses `.GetStore().GetByKey("namespace/name")`. @mhenriks will appreciate you lgtm if it looks good